packages/block-library/src/navigation-link/edit.js

```diff
@@ -160,6 +161,20 @@ function NavigationLinkEdit( {
 						<LinkControl
 							className="wp-block-navigation-link__inline-link-input"
 							value={ link }
+							initialSuggestions={ () => {
```

Comment:
I believe the LinkControl has already a built-in mechanism to fetch suggestions. Is calling the suggestion fetcher with an empty string equivalent to the result of this function? Should we instead do that in the component instead of having to duplicate it here?

Potentially this could become just something like `showSuggestionsOnOpen` or something like that?

Reply:
> Is calling the suggestion fetcher with an empty string equivalent to the result of this function?

In short no. Please see [my original research](https://github.com/WordPress/gutenberg/issues/18899#issuecomment-571520975) which shows that this approach won't work. 

I'm loathed to add another API but it seems like a legit use case that I can imagine being required in other scenarios.

Reply:
So based on your comment, it seems that the main blockers are coming from URLInput? Am I reading that right?

I wonder if we should get rid of that component entirely and just reimplement it in LinkControlSearch 🤷‍♂

Reply:
Yeh all the complexity in this component comes form `URLInput`. Personally I think what we need is a lower-level component which implements the UI of the search which can then be used in `LinkControl`. `URLInput` is too opinionated on its original use-case which is understandable but is starting to cause issues.

I just need help making the call about when it's viable to commit time/effort to it. If you think the time has come then that's strong evidence for me to present to stakeholders.

Reply:
I went ahead and created [an Issue for the potential refactor here](https://github.com/WordPress/gutenberg/issues/19523).

Reply:
I've done a little bit of work on tidying it up—making a presentational version of URLInput. I'll try to get a PR up, I need to tidy/fix a few things first that I broke along the way.